src/rxjs/atomWithObservable.ts

```diff
@@ -43,11 +71,23 @@ export function atomWithObservable<TData>(
         setResult({ error })
       }
     }
-    observable.pipe(first()).toPromise().then(dataListener).catch(errorListener)
+    let subscription: Subscription | null = null
+    subscription = observable.subscribe((data) => {
+      dataListener(data)
+      if (subscription && !setResult) {
+        subscription.unsubscribe()
+        subscription = null
+      }
+    }, errorListener)
+    if (!resolve) {
+      subscription.unsubscribe()
+      subscription = null
+    }
```

Comment:
the reason this would be required is subscription callback is invoked in sync, but then it's already unsubscribed.
hm, what's the intention behind this?